users/tests/tests.py

```diff
@@ -34,9 +44,43 @@ def test_redirection_on_register_successful(self):
 
     def test_register_successful(self):
         response = self.client.post(reverse('users:register'), {'username': 'user_1', 'password1': 'pass1234', 'password2': 'pass1234'}, follow=True)
-        self.assertIn('You successfully created a new account!', response.content)
+        self.assertIn(UserRegistration.success_message, response.content)
 
     def test_register_unsuccessful(self):
         UserFactory(username='existing_user')
         response = self.client.post(reverse('users:register'), {'username': 'existing_user', 'password1': 'pass1234', 'password2': 'pass1234'})
         self.assertIn('A user with that username already exists.', response.content)
+
+
+class UserLoginIntegrationTests(TestCase, SimpleTestCase):
+    def test_redirection_on_login_successful(self):
+        UserFactory(username='existing_user')
+        response = self.client.post(reverse('users:login'), {'username': 'existing_user', 'password': 'pass1234'})
+        self.assertEqual(response.status_code, 302)
+
+    def test_login_successful(self):
+        UserFactory(username='existing_user')
+        response = self.client.post(reverse('users:login'), {'username': 'existing_user', 'password': 'pass1234'})
+        self.assertRedirects(response, reverse('users:user-profile'))
+
+    def test_login_unsuccessful(self):
+        response = self.client.post(reverse('users:login'), {'username': 'non_existing_user', 'password': 'pass1234'})
+        self.assertIn('Please enter a correct username and password. Note that both fields may be case-sensitive.', response.content)
+
+
+class UserProfileIntegrationTests(TestCase):
+    def test_user_profile_page_with_authenticated_user(self):
+        UserFactory(username='existing_user')
+        self.client.post(reverse('users:login'), {'username': 'existing_user', 'password': 'pass1234'})
+        response = self.client.get(reverse('users:user-profile'))
+        self.assertIn('You are logged in!', response.content)
```

Comment:
Were you not able to use `assertTemplateUsed` instead of checking the content of the response?
This question goes to all the places you made such an assertion.